src/proof_of_possession.rs

```diff
@@ -148,18 +152,21 @@ impl ProofOfPossession {
     ) -> Result<(), OIDCError> {
         let now = Utc::now();
 
+        let nbf_tolerance = params.nbf_tolerance.unwrap_or_else(Duration::zero);
+        let exp_tolerance = params.exp_tolerance.unwrap_or_else(Duration::zero);
+
         if let Some(not_before) = self.body.not_before.clone() {
             let nbf = not_before.try_into()?;
-            if now < nbf {
+            if (now + nbf_tolerance) < nbf {
```

Comment:
I had to think about whether the operation should be addition or subtraction and whether nbf_tolerance is a positive or negative duration, but I think I've convinced myself this is correct. Looking at the test code helped.